roofit/hs3/src/HistFactoryJSONTool.cxx

```diff
@@ -0,0 +1,263 @@
+}
+
+
+void RooStats::HistFactory::JSONTool::Export(const RooStats::HistFactory::Channel& c, JSONNode& ch) const {
+  ch.set_map();
+  ch["type"] << "histfactory";      
+  
+  auto& staterr = ch["statError"];
+  staterr.set_map();
+  staterr["relThreshold"] << c.GetStatErrorConfig().GetRelErrorThreshold();      
+  staterr["constraint"] <<RooStats::HistFactory::Constraint::Name(c.GetStatErrorConfig().GetConstraintType());
+  auto& stack = staterr["stack"];
+  stack.set_seq();
+  
+  auto& samples = ch["samples"];
+  samples.set_map();
+  for(const auto& s:c.GetSamples()){
+    auto& sample = samples[s.GetName()];
+    this->Export(s,sample);
+    auto& ns = sample["namespaces"];
+    ns.set_seq();
+    ns.append_child() << c.GetName();
+    stack.append_child() << s.GetName();
+  }
+}
+
+void RooStats::HistFactory::JSONTool::Export(JSONNode& n) const {
+  for(const auto& ch:this->_measurement->GetChannels()){
+    if(!ch.CheckHistograms()) throw std::runtime_error("unable to export histograms, please call CollectHistograms first");
+  }
+
+  //   auto parlist = n["variables"];
+  //   parlist |= c4::yml::MAP;
+
+  auto& pdflist = n["pdfs"];
+  pdflist.set_map();
+
+  // collect information
+  std::map<std::string,RooStats::HistFactory::Constraint::Type> constraints;
+  std::map<std::string,NormFactor > normfactors;
+  for(const auto& ch:this->_measurement->GetChannels()){
+    for(const auto&s: ch.GetSamples()){
+      for(const auto& sys:s.GetOverallSysList()){
+        constraints[sys.GetName()] = RooStats::HistFactory::Constraint::Gaussian;
+      }
+      for(const auto& sys:s.GetHistoSysList()){
+        constraints[sys.GetName()] = RooStats::HistFactory::Constraint::Gaussian;
+      }
+      for(const auto& sys:s.GetShapeSysList()){
+        constraints[sys.GetName()] = sys.GetConstraintType();
+      }
+      for(const auto& norm:s.GetNormFactorList()){
+        normfactors[norm.GetName()] = norm;
+      }
+    }
+  }
+
+  // parameters
+  
+  //  auto lumi = parlist["Lumi"];
+  //  lumi |= c4::yml::MAP;  
+  //  lumi["value"] << this->_measurement->GetLumi();  
+  //  lumi["relErr"] << this->_measurement->GetLumiRelErr();
+
+  //   for(const auto& par:this->_measurement->GetPOIList()){
+  //     auto node = parlist[c4::to_csubstr(par)];
+  //     node |= c4::yml::MAP;    
+  //     if(this->_measurement->GetParamValues().find(par) != this->_measurement->GetParamValues().end()){
+  //       node["value"] << this->_measurement->GetParamValues().at(par);
+  //     }    
+  //   }
+  //   for(const auto& par:this->_measurement->GetParamValues()){
+  //     auto node = parlist[c4::to_csubstr(par.first)];
+  //     node |= c4::yml::MAP;    
+  //     node["value"] << par.second;
+  //     if(std::find(this->_measurement->GetConstantParams().begin(),this->_measurement->GetConstantParams().end(),par.first) != this->_measurement->GetConstantParams().end()){
+  //       node["const"] << 1;
+  //     }
+  //   }
+  //   
```

Comment:
This will presumably disappear?

Reply:
yes, thanks for the pointer, I overlooked this. I'll get rid of it.